### PR TITLE
[luci/pass] Introduce GeluPattern2

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This PR introduces GeluPattern2.

for issue: https://github.com/Samsung/ONE/issues/11002
from draft: #11016

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

File: compiler/luci/pass/src/FuseGeluPass.cpp [hunk 1]
```diff
@@ -179,13 +273,13 @@ luci::CircleGelu *FuseGelu::create_gelu(loco::Graph *graph)
   gelu->features(_p->_ifm);
   // TODO Support approximate = True pattern
   gelu->approximate(false);
-  gelu->name(_p->_mul_half->name() + "_gelu");
+  gelu->name(_p->_pattern_last_node->name() + "_gelu");
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseGeluPass.cpp:276]: This change doesn't seem to be related with `GeluPattern2`, can you describe how it is?

File: compiler/luci/pass/src/FuseGeluPass.cpp [hunk 2]
```diff
@@ -179,13 +273,13 @@ luci::CircleGelu *FuseGelu::create_gelu(loco::Graph *graph)
...
   return gelu;
 }
 
 void FuseGelu::apply()
 {
-  auto graph = _p->_mul_half->graph();
+  auto graph = _p->_pattern_last_node->graph();
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseGeluPass.cpp:282]: ditto

File: compiler/luci/pass/src/FuseGeluPass.cpp [hunk 3]
```diff
@@ -196,7 +290,7 @@ void FuseGelu::apply()
 
   luci::add_origin(gelu, luci::composite_origin(origin_vec));
 
-  replace(_p->_mul_half).with(gelu);
+  replace(_p->_pattern_last_node).with(gelu);
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseGeluPass.cpp:293]: ditto

File: compiler/luci/pass/src/FuseGeluPass.cpp (hunk 1 above)
Comment (BalyshevArtem) [reply compiler/luci/pass/src/FuseGeluPass.cpp]: Removed it

File: compiler/luci/pass/src/FuseGeluPass.cpp [hunk 4]
```diff
@@ -216,6 +310,9 @@ bool fuse_gelu(luci::CircleMul *mul)
     fuse.apply();
     return true;
   }
+
+  GeluPattern2 pattern2(mul);
+  (void)pattern2;
```
Comment (BalyshevArtem) [inline compiler/luci/pass/src/FuseGeluPass.cpp:315]: to avoid define but not used error

**Review by seanshpark — APPROVED** [review]
LGTM
